src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -258,6 +260,13 @@ export class FetchPlugin extends MonkeyPatched<Window, 'fetch'> {
         this.context.record(HTTP_EVENT_TYPE, httpEvent);
     };
 
+    /** Manually updates the http duration and returns the end time in seconds */
+    private updateDurationManually(httpEvent: HttpEvent): number {
+        const endTime = Date.now();
+        httpEvent.duration = endTime - httpEvent.startTime!;
+        return endTime / 1000;
+    }
```

Comment:
suggestion: Seems like we've combined a getter and setter method into one method. Can we clarify what singular action this method should do and refactor accordingly? Also want to note that we're setting end time with `ms` as the unit but returning the value in seconds which may be confusing at the first glance. Could we unify to use one unit and do the unit conversion outside this method?

Reply:
Changes:
* Specificed ms / sec in all timestamp vars for clarity 
* Removed `setDurationManually()`

My intention behind `setDurationManually()` was code reuse, but I can see how it broke SOLID and was confusing.